### PR TITLE
[spec] [chapter-20] Create bibliography for Related Documents

### PR DESCRIPTION
**Pull request by hussainnm**
* Add numbers as label for bibliography

### CODE REVIEW

File: spec/src/main/asciidoc/core/RelatedDocuments.adoc [hunk 1]
```diff
@@ -1,137 +1,137 @@
 [[a9849]]
+[bibliography]
 == Related Documents
 
-Enterprise JavaBeans™, version 3.2. (this
+- [[[a9850, 1]]] Enterprise JavaBeans™, version 3.2. (this
```
Comment (kwsutter) [inline spec/src/main/asciidoc/core/RelatedDocuments.adoc]: All of these spec references should be pointing at the Jakarta EE location now...  For example, https://jakarta.ee/specifications/enterprise-beans/3.2/

File: spec/src/main/asciidoc/core/RelatedDocuments.adoc [hunk 2]
```diff
@@ -1,137 +1,137 @@
...
 specification) _http://jcp.org/en/jsr/detail?id=3_ 45.
 
-[[a9851]]Java™ Persistence API,
+- [[[a9851, 2]]] Java™ Persistence API,
```
Comment (kwsutter) [inline spec/src/main/asciidoc/core/RelatedDocuments.adoc]: We should be replacing the `Java `references with `Jakarta` (in most cases).  Similar thing for `JavaBeans`.  The context surrounding the usage may affect these type of changes, but you'd be safer doing a blanket change and then correcting the individual cases.

**Comment by hussainnm** [comment]
Before I make the changes, I would like to put my thoughts out:

Existing | New
-------- | -----
[[[a9850, 1]]] Enterprise JavaBeans™, version 3.2. (this specification) _http://jcp.org/en/jsr/detail?id=345_. | [[[a9850, 1]]] Jakarta Enterprise Beans (Core) 4.0. (this specification)  _https://jakarta.ee/specifications/enterprise-beans/4.0/_.
[[[a9851, 2]]] Java™ Persistence API, version 2.1. _http://jcp.org/en/jsr/detail?id=338_ | [[[a9851, 2]]] Jakarta Persistence 3.0. _https://jakarta.ee/specifications/persistence/3.0/_
[[[a9857, 8]]] Java™ Transaction API, version 1.2 (JTA). http://jcp.org/en/jsr/detail?id=907. | [[[a9857, 8]]] Jakarta Transactions 2.0. _https://jakarta.ee/specifications/transactions/2.0/_.
[[[a9861, 12]]] Java™ Platform, Enterprise Edition Specification Version 7 (Java EE). _http://jcp.org/en/jsr/detail?id=342_ | [[[a9861, 12]]] Jakarta EE Platform 9. _https://jakarta.ee/specifications/platform/9/_.
[[[a9862, 13]]] Java™ Message Service, version 2.0 (JMS). http://jcp.org/en/jsr/detail?id=343. | [[[a9862, 13]]] Jakarta Messaging 3.0. _https://jakarta.ee/specifications/messaging/3.0/_.
[[[a9863, 14]]] Java EE™ Connector Architecture, version 1.7 (Connector). _http://jcp.org/en/jsr/detail?id=322_ . | [[[a9863, 14]]] Jakarta Connectors 2.0. _https://jakarta.ee/specifications/connectors/2.0/_ .
[[[a9873, 24]]] Java™ API for XML-based RPC, version 1.1 (JAX-RPC). _http://jcp.org/en/jsr/detail?id=101._ | [[[a9873, 24]]] Jakarta XML RPC 1.1. _https://jakarta.ee/specifications/xml-rpc/1.1/_.
[[[a9878, 29]]] Web Services Metadata for the Java Platform, version 2.1. _http://jcp.org/en/jsr/detail?id=181._ | [[[a9878, 29]]] Jakarta Web Services Metadata 3.0. _https://jakarta.ee/specifications/web-services-metadata/3.0/_.
[[[a9881, 31]]] _Java™ API for XML-based Web Service, version 2.2 (JAX-WS). http://jcp.org/en/jsr/detail?id=224._ | [[[a9881, 31]]] Jakarta XML Web Services 3.0. _https://jakarta.ee/specifications/xml-web-services/3.0/_
[[[a9886, 36]]] Managed Beans, version 1.0 (Managed Beans). _http://jcp.org/en/jsr/detail?id=316._ | [[[a9886, 36]]] Jakarta Managed Beans 2.0. _https://jakarta.ee/specifications/managedbeans/2.0/_.
[[[a9887, 37]]] Interceptors, version 1.2. _http://jcp.org/en/jsr/detail?id=318._ | [[[a9887, 37]]] Jakarta Interceptors 2.0. _https://jakarta.ee/specifications/interceptors/2.0/_.
[[[a9888, 38]]] Contexts and Dependency Injection for the Java EE Platform 1.1 (CDI specification) http://jcp.org/en/jsr/detail?id=346. | [[[a9888, 38]]] Jakarta Context Dependency Injection 3.0 https://jakarta.ee/specifications/cdi/3.0/.
[[[a9889, 39]]] The Java API for RESTful Web Services, version 2.0 (JAX-RS) http://jcp.org/en/jsr/detail?id=339. | [[[a9889, 39]]] Jakarta RESTful Web Services 3.0 https://jakarta.ee/specifications/restful-ws/3.0/.
[[[a9890, 40]]] EJB 3.2 Optional Features _http://jcp.org/en/jsr/detail?id=345_ | [[[a9890, 40]]] Jakarta Enterprise Beans (Optional) 4.0. _https://jakarta.ee/specifications/enterprise-beans/4.0/_.

Regarding JSR-109, since it is not in Jakarta EE we can refer to the JCP document but I believe the concepts are implemented in Glassfish.
[[[a9879, 30]]] Web Services for Java EE, version 1.3. _http://jcp.org/en/jsr/detail?id=109._
will be updated to 
[[[a9879, 30]]] Web Services for Java EE™, version 1.4. _http://jcp.org/en/jsr/detail?id=109._

Similarly Java Transaction Service is also not in Jakarta EE.
[[[a9858, 9]]] Java™ Transaction Service, version 1.0 (JTS). http://www.oracle.com/technetwork/java/javaee/jts-spec095-1508547.pdf.

**Comment by tkburroughs** [comment]
The current changes in this PR move from Java to Jakarta, where appropriate, but don't have the spec version updates yet, like 3.2-4.0, etc.  Were you going to make those changes with this PR, or defer to a latter PR?  The mapping from existing->new in your table lgtm.

**Comment by hussainnm** [comment]
I will defer the version updates to a later PR which will incorporate other changes (removals). The Introduction, Related Documents and Revision History needs to be revised and can be done in a separate PR.

**Pull request merged**